pkg/gitops/flux/files.go

```diff
@@ -85,7 +88,13 @@ func (g *FileGenerator) WriteEksaFiles(clusterSpec *cluster.Spec, datacenterConf
 		return err
 	}
 
-	if err := g.WriteEksaKustomization(); err != nil {
+	if hardwareCSVPath != "" {
```

Comment:
If the caller knows whether or not a hardware CSV is available, would it make sense to just have that decide to invoke `WriteHardware` instead of pushing it through to here?

Reply:
I think that maintaining the write in this method is ultimately better, since it's responsible for writing all the files under the `eksa-system` folder.